props/saber_BC_buttons.h

```diff
@@ -477,738 +1478,1227 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+                   << (target_blade[BC_MAIN_BLADE] ? "SECOND" : "MAIN")
+                   << " blade\n";
+      if (isPointingUp() || muted_) {
+        FastOn(EffectLocation(0, target_blade));
+     } else {
+        On(EffectLocation(0, target_blade));
       }
-      return true;
+    } else {
+      PVLOG_NORMAL << "** Turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade\n";
+      SaberBase::TurnOn(EffectLocation(0, target_blade));
+    }
+  }
 
-// Enter / Exit Volume MENU
-#ifndef NO_VOLUME_MENU
+  void TurnBladeOff(BladeSet target_blade) {
+    SaberBase::OffType off_type = isPointingUp() ? SaberBase::OFF_FAST : SaberBase::OFF_NORMAL;
+    PVLOG_DEBUG << "***** Off type: " << (off_type == SaberBase::OFF_FAST ? "OFF_FAST" : "OFF_NORMAL") << "\n";
+
+    // Check if this is the only blade ON (of MAIN or SECOND blades)
+    if ((SaberBase::OnBlades() & ~target_blade & controlled_blades_).off()) {
+      // Other blade is not on, so just do normal Off() with appropriate off_type
+      PVLOG_NORMAL << "** Turning OFF all blades\n";
+      Off(off_type);
+      muted_ = false;
+    } else {
+      // Only Turn OFF this blade, leave the other one ON.
+      PVLOG_NORMAL << "** Turning OFF only the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND") 
+                   << " blade\n";
+      SaberBase::TurnOff(off_type, EffectLocation(0, target_blade));
+    }
+    battle_mode_ = false;
+    spam_blast_ = false;
+    saber_off_time_ = millis();
+  }
+
+  void TurnMainBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_mainBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        PVLOG_NORMAL << "** Main Blade Turned ON Muted\n";
+      }
+    }
+  }
+
+  void TurnSecondBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_secondBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_SECOND_BLADE_SET);
+        PVLOG_NORMAL << "** Second Blade Turned ON Muted\n";
+      }
+    }
+  }
+
+  void TurnDualBladesOn() {
+    TurnOnHelper();
+    PVLOG_NORMAL << "**** Dual Blades Turned ON\n";
+  }
+
+  void TurnDualBladesOff() {
+    TurnOffHelper();
+    PVLOG_NORMAL << "** Dual Blades Turned OFF\n";
+  }
+
+  // Determine the active blade based on x-axis motion - for thrust effects
+  void GetThrustBladeLocation() {
+    mss = fusor.mss();
+    if (mss.x < 14) {
+        thrusting_blade_ = BC_MAIN_BLADE_SET;
+    } else if (mss.x > -14) {
+        thrusting_blade_ = BC_SECOND_BLADE_SET;
+    }
+    thrust_location = EffectLocation(0, thrusting_blade_);
+  }
+
+  bool IsStab(const Vec3& diff) override {
```

Comment:
This seems to get called a bunch, and it's fluctuating a lot with what it sees.
It's not really trustworthy to choose the correct blade, sort of a 50/50 chance in my testing.
Also, it seems for some reason that this as well as Thrust calculations work better when the target blade is angled slightly down?
Maybe there's some filtering that could be applied to smooth out the readings so that a better decision could be made?

Reply:
When this is used for stab, only the return value at the time that the clash is triggered actually counts. All the other calculations don't do anything.

However, since you are changing forward_stab_ inside the function, and this function may be called a bunch more times before the event is processed, you get bad values.

At least I think that's what's going on.

Breaking out IsStab() might not have been the right way to do this, because we don't actually have all the information we need in this function now...

Reply:
> When this is used for stab, only the return value at the time that the clash is triggered actually counts. All the other calculations don't do anything.
> 
> However, since you are changing forward_stab_ inside the function, and this function may be called a bunch more times before the event is processed, you get bad values.
> 
> At least I think that's what's going on.
> 
> Breaking out IsStab() might not have been the right way to do this, because we don't actually have all the information we need in this function now...

This was correct. So I reverted IsStab() and restored DoAccel() completely in prop_base, see https://github.com/profezzorn/ProffieOS/pull/692.
Then I just made a full DoAccel() override in my prop, and added the foward_stab_ detection in the right place.
See latest update.
Works in testing.